lib/matplotlib/tests/test_dates.py

```diff
@@ -511,6 +509,113 @@ def _create_auto_date_locator(date1, date2):
         assert strings == expected
 
 
+def test_concise_formatter_formats():
+    formats = ['%Y', '%m/%Y', 'day: %d',
+               '%H hr %M min', '%H hr %M min', '%S.%f sec']
+
+    def _create_auto_date_locator(date1, date2):
+        fig, ax = plt.subplots()
+
+        locator = mdates.AutoDateLocator(interval_multiples=True)
+        formatter = mdates.ConciseDateFormatter(locator, formats=formats)
+        ax.yaxis.set_major_locator(locator)
+        ax.yaxis.set_major_formatter(formatter)
+        ax.set_ylim(date1, date2)
+        fig.canvas.draw()
+        sts = [st.get_text() for st in ax.get_yticklabels()]
+        return sts
+
+    d1 = datetime.datetime(1997, 1, 1)
+    results = (
+        [datetime.timedelta(weeks=52 * 200), [str(t) for t in range(1980,
+         2201, 20)]],
+        [datetime.timedelta(weeks=52), [
+            '1997', '02/1997', '03/1997', '04/1997', '05/1997', '06/1997',
```

Comment:
```suggestion
            '01/1997', '02/1997', '03/1997', '04/1997', '05/1997', '06/1997',
```
This bug is not failing the test. Why? Something is fishy here.

Reply:
The `zero_formats` parameter is left at default in this test. For the beginning of a year, the default is not to have the month. The above is expected behavior.

Reply:
@tillahoffmann Perhaps the choice of `formats` params is a bit odd in the context of the default `zero_formats`, but my reasoning was to keep tests for each set of parameters separate. I can change that if it's preferable.

Reply:
The output you suggest is what you get if you set,
`formats = ['%m/%Y', '%m/%Y', 'day: %d',
               '%H hr %M min', '%H hr %M min', '%S.%f sec']`
however, the original is indeed what you get with formats set the way it is in line 512.